src/vs-bicep/Bicep.VSLanguageServerClient.IntegrationTests/Utilities/WaitForExtensions.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Microsoft.Test.Apex.Services;
+
+namespace Bicep.VSLanguageServerClient.IntegrationTests.Utilities
+{
+    public static class WaitForExtensions
+    {
+        public static void IsTrue(Func<bool> predicate, TimeSpan timeout, Func<string>? onTimeout = null, string? conditionDescription = null)
+        {
+            bool result = WaitFor.TryIsTrue(predicate, timeout);
+
+            if (!result)
+            {
+                conditionDescription ??= "the condition to become true";
+
+                if (onTimeout != null)
+                {
+                    // Call the timout callback.
+                    conditionDescription = string.Concat(conditionDescription, "\r\n onTimeout result: ", onTimeout(), "\r\n");
+                }
+
+                // Determine if doubling the timout would cause the predicate to succeed
+                result = WaitFor.TryIsTrue(predicate, timeout);
+
+                if (!result)
+                {
+                    conditionDescription = conditionDescription + " (expired even with doubled timeout length)";
+                }
+                else
+                {
+                    conditionDescription = conditionDescription + " (*didn't expire* with doubled timeout length)";
+                }
+
+                // This will always throw as it's always expired in this codepath
+                throw new TimeoutException(string.Format(CultureInfo.InvariantCulture, "Timed out after {0} waiting for {1}", timeout, conditionDescription));
+            }
+        }
+
+        public static void AreEqual<T>(Func<T> getActual, T expected, TimeSpan timeout, Action<T>? onTimeout = null)
```

Comment:
This can't be implemented using IsTrue?

Reply:
Updated to use IsTrue instead:
https://github.com/Azure/bicep/commits/dev/bhsubra/Bicep_VSIntegration